tensorboard/components/tf_backend/router.ts

```diff
@@ -86,42 +87,55 @@ export function setRouter(router: Router): void {
   _router = router;
 }
 
-function createProdPath(pathPrefix: string, path: string,
+function createProdDataPath(dataDir: string, route: string,
     ext: string, params?: URLSearchParams): string {
-  const url = new URL(window.location.origin);
-  // Use URL to normalize pathPrefix with leading slash and without.
-  url.pathname = pathPrefix + path;
-  if (params) url.search = params.toString();
-  return url.pathname + url.search;
+  let relativePath = dataDir + route;
+  if (params) {
```

Comment:
Note (not a big deal): if I read this correctly, the behavior is
different when `params` is `new URLSearchParams()` vs. `undefined`: in
the former case you get URLs like `/data/foo?` and `/data/foo?bar=1&`,
while in the latter case you get `/data/foo` and `/data/foo?bar=1`. To
my knowledge, these should be handled identically by the backend, so I’m
fine with the code as is, even though it would IMHO be more sensible to
write as

```typescript
function createProdDataPath(
  dataDir: string,
  route: string,
  ext: string,
  params: URLSearchParams = new URLSearchParams()
): string {
  let relativePath = dataDir + route;
  if (String(params)) {
    const delimiter = route.includes("?") ? "&" : "?";
    relativePath += delimiter + String(params);
  }
  return relativePath;
}
```